experiments/commands.py

```diff
@@ -0,0 +1,234 @@
+
+And get the following commands printed out:
+
+python -m imitation.scripts.train_adversarial airl \
+    --capture=sys --name=run0 \
+    --file_storage=output/sacred/$USER-cmd-run0-airl-0-a3531726 \
+    with ../benchmarking/example_airl_seals_walker_best_hp_eval.json \
+    seed=0 logging.log_root=output
+
+python -m imitation.scripts.train_adversarial gail \
+    --capture=sys --name=run0 \
+    --file_storage=output/sacred/$USER-cmd-run0-gail-0-a1ec171b \
+    with ../benchmarking/example_gail_seals_walker_best_hp_eval.json \
+    seed=0 logging.log_root=output
+
+We can execute commands in parallel by piping them to GNU parallel:
+
+python commands.py ... | parallel -j 8
+
+If the --remote flag is enabled, then the program prints out commands
+to run training scripts in containers on the Hofvarpnir cluster.
+
+For example, we can run:
+
+python commands.py \
+    --name=run0 \
+    --cfg_pattern=../benchmarking/example_bc_seals_half_cheetah_best_hp_eval.json \
+    --output_dir=/data/output \
+    --remote
+
+And get the following command printed out:
+
+ctl job run --name $USER-cmd-run0-bc-0-72cb1df3 \
+    --command python\\ -m\\ imitation.scripts.train_imitation\\ bc\\ \
+    --capture=sys\\ --name=run0\\ \
+    --file_storage=/data/output/sacred/$USER-cmd-run0-bc-0-72cb1df3\\ \
+    with\\ \
+    /data/imitation/benchmarking/example_bc_seals_half_cheetah_best_hp_eval.json\\ \
+    seed=0\\ logging.log_root=/data/output --container hacobe/devbox:imitation \
+    --login --high-priority --force-pull --never-restart
+"""
+import argparse
+import glob
+import os
+import zlib
+
+_ALGO_NAME_TO_SCRIPT_NAME = {
+    "bc": "train_imitation",
+    "dagger": "train_imitation",
+    "airl": "train_adversarial",
+    "gail": "train_adversarial",
+}
+
+_CMD_ID_TEMPLATE = "$USER-cmd-{name}-{algo_name}-{seed}-{cfg_id}"
+
+_TRAIN_CMD_TEMPLATE = (
+    "python -m imitation.scripts.{script_name} {algo_name} "
+    "--capture=sys --name={name} --file_storage={file_storage} "
+    "with {cfg_path} seed={seed} logging.log_root={log_root}"
+)
+
+_HOFVARPNIR_CLUSTER_CMD_TEMPLATE = (
+    "ctl job run --name {name} --command {command} --container {container} "
+    "--login --high-priority --force-pull --never-restart"
+)
+
+
+def _get_algo_name(cfg_file: str) -> str:
+    algo_names = set()
+    for key in _ALGO_NAME_TO_SCRIPT_NAME:
+        if cfg_file.find("_" + key + "_") != -1:
+            algo_names.add(key)
+
+    if len(algo_names) == 0:
+        raise ValueError("Unable to find algo_name in cfg_file: " + cfg_file)
+
+    if len(algo_names) >= 2:
+        raise ValueError("algo_name is ambiguous in cfg_file: " + cfg_file)
+
+    algo_name = algo_names.pop()
+    return algo_name
+
+
+def _get_cfg_id(cfg_path: str) -> str:
+    checksum = zlib.adler32(cfg_path.encode())
+    checksum_hex = hex(checksum)
+    assert checksum_hex.startswith("0x")
+    return checksum_hex[2:]
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
```

Comment:
Nit: can we make this call a main method? Anything run at module-level pollutes the global namespace. Doesn't matter too much, but does slightly raise risk of other methods accidentally using a non-local value.

Reply:
By the looks of it could probably also split into a parsing method, and a method that does the bulk of the logic.